### ISSUE TITLE
Inconsistent off-by-one errors with column numbers

### ISSUE THREAD

**Issue by gcochard** [eslint/eslint#3231]
Reported after scrooloose/syntastic#1494 was closed. I have replicated the behavior with eslint `v1.0.0-rc-3` and `v1.0.0`.
```bash
0:11:[greg:~/Dev/github/spectcl]$ eslint lib/spectcl.js 

lib/spectcl.js
  10:44  error  Extra semicolon  semi

✖ 1 problem (1 error, 0 warnings)

1:12:[greg:~/Dev/github/spectcl]$ head lib/spectcl.js | tail -1
var spawn = require('child_process').spawn;
0:13:[greg:~/Dev/github/spectcl]$ eslint --version
v1.0.0
```

When using vim, it highlights the column **after** the semicolon as shown in the screenshot below:
<img width="332" alt="screen shot 2015-07-31 at 2 23 24 pm" src="https://cloud.githubusercontent.com/assets/2836237/9018094/5b09e4b2-3790-11e5-9521-a4f1d678984b.png">

Note that I have no trailing spaces in the above example.

Below, I have a trailing space on line 3, and I have `:set list` turned on in vim so that you may see the space.

<img width="443" alt="screen shot 2015-07-31 at 2 39 35 pm" src="https://cloud.githubusercontent.com/assets/2836237/9018265/0d416f64-3792-11e5-8781-a4bf4340a36d.png">

As you can see, trailing spaces are reported to be one column greater than they should be, same with semi, and the indent rule sometimes shows the correct column and sometimes shows off-by-one. I have tried with `} else { ...` on line 18 below, but still get the same off-by-one, so it's not due to the error being on the last character of a line.
Code:
```js
'use strict'
/*
 * spectcl.js: Top-level include for the `spectcl` module. 
 *
 * (C) 2015, Greg Cochard, Ryan Milbourne
 * (C) 2011, Elijah Insua, Marak Squires, Charlie Robbins.
 *
 */

var spawn = require('child_process').spawn;
  var util = require('util')
var AssertionError = require('assert').AssertionError
var EventEmitter = require('events').EventEmitter

function testExpectation(data, expectation) {
    if (util.isRegExp(expectation)) {
        return expectation.test(data)
  }
```

Errors:
```
lib/spectcl.js|3 col 60 error| Trailing spaces not allowed. (no-trailing-spaces)                                                 
lib/spectcl.js|10 col 44 error| Extra semicolon. (semi)
lib/spectcl.js|11 col 3 error| Expected indentation of 0 characters. (indent)
lib/spectcl.js|18 col 4 error| Expected indentation of 4 characters. (indent)
```

Config file follows:

```json
{
    "env": {"node": true},
    "globals": {},
    "extends": "eslint:recommended",
    "rules": {
      "no-extra-parens": 2,
      "no-unexpected-multiline": 2,
      "valid-jsdoc": 2,
      "default-case": 2,
      "dot-notation": 2,
      "eqeqeq": 2,
      "no-alert": 2,
      "no-caller": 2,
      "no-else-return": 2,
      "no-empty-label": 2,
      "no-eval": 2,
      "no-extend-native": 2,
      "no-floating-decimal": 2,
      "no-implied-eval": 2,
      "no-iterator": 2,
      "no-lone-blocks": 2,
      "no-loop-func": 2,
      "no-multi-spaces": 2,
      "no-multi-str": 2,
      "no-native-reassign": 2,
      "no-new-func": 2,
      "no-new-wrappers": 2,
      "no-new": 2,
      "no-octal-escape": 2,
      "no-proto": 2,
      "no-return-assign": 2,
      "no-self-compare": 2,
      "no-throw-literal": 2,
      "no-unused-expressions": 2,
      "no-warning-comments": 2,
      "no-with": 2,
      "radix": 2,
      "strict": [2, "global"],
      "no-shadow-restricted-names": 2,
      "no-shadow": 2,
      "no-undef-init": 2,
      "no-use-before-define": 2,
      "callback-return": 2,
      "handle-callback-err": 2,
      "no-mixed-requires": 2,
      "no-new-require": 2,
      "no-path-concat": 2,
      "no-process-exit": 2,
      "no-sync": 2,
      "camelcase": 2,
      "consistent-this": [2, "self"],
      "indent": 2,
      "linebreak-style": 2,
      "new-cap": 2,
      "new-parens": 2,
      "no-array-constructor": 2,
      "no-fallthrough": 1,
      "no-multiple-empty-lines": [2, {"max": 2}],
      "no-nested-ternary": 2,
      "no-new-object": 2,
      "no-trailing-spaces": 2,
      "quotes": [2, "single", "avoid-escape"],
      "semi": [2, "never"],
      "space-return-throw-case": 2,
      "spaced-comment": [2, "always", {"exceptions": ["=","-"]}]
    }
}
```

**Comment by btmills** [comment]
Had the vim integration been updated for the 1-based column indexing we switched to in 1.0?

**Comment by gcochard** [comment]
Yes, I confirmed that the vim integration was working properly.

**Comment by nzakas** [comment]
Are you sure? It looks like it hasn't been updated yet:

https://github.com/scrooloose/syntastic/blob/master/syntax_checkers/javascript/eslint.vim#L52

**Comment by nzakas** [comment]
Ooh, nevermind, I see what they're doing.

Can someone confirm if this behavior is seen on the command line?

**Comment by gcochard** [comment]
The issue post shows the direct eslint output on the command line, at the very top of the post. If you look at the output and then count the characters in the line, there are 43, not 44.

It could be that the behavior is expected, as it is warning on the newline character, not on the semicolon. Thoughts on that?

**Comment by nzakas** [comment]
This could just be a quirk of how the rules are written. For instance:
https://github.com/eslint/eslint/blob/master/lib/rules/semi.js#L27

If you are disallowing semicolons, then the last token would be the semicolon and the end location is the spot right after it. It could be that we are doing something similar in a handful of rules.

We should double check semi and no-trailing-spaces to make sure the locations are correct.
